### PR TITLE
SYMPOWER-3065 | Expand simple logging listener signature to include raw message format

### PR DESCRIPTION
**Pull request by tonisojandu-sympower**

### CODE REVIEW

File: slf4j-simple/src/main/java/org/slf4j/impl/SimpleLogger.java [hunk 1]
```diff
@@ -314,27 +314,31 @@ private static int stringToLevel(String levelStr) {
...
     private static void replayBufferedEvents(SimpleLogListener listener) {
         for(int i = 0; i < PREINIT_BUFFER.size(); i++) {
             SimpleLogEvent event = (SimpleLogEvent) PREINIT_BUFFER.elementAt(i);
-            event.logger.logImpl(listener, event.timestamp, event.level, event.threadName, event.message, event.t);
+            event.logger.logImpl(listener, event.timestamp, event.level, event.threadName, event.messageFormat, event.message, event.t);
         }
     }
 
-    private void logImpl(SimpleLogListener listener, long timestamp, int level, String threadName, String message, Throwable t) {
+    private void logImpl(SimpleLogListener listener, long timestamp, int level, String threadName,  String messageFormat, String message,Throwable t) {
```
Comment (neeme-praks-sympower) [inline slf4j-simple/src/main/java/org/slf4j/impl/SimpleLogger.java]:
```suggestion
    private void logImpl(SimpleLogListener listener, long timestamp, int level, String threadName, String messageFormat, String message, Throwable t) {
```

File: slf4j-simple/src/test/java/org/slf4j/impl/SimpleLoggerTest.java [hunk 2]
```diff
@@ -75,4 +82,46 @@ public void loggerNameWithOneDot_NoSetLevel() {
...
+        SimpleLogger simpleLogger = new SimpleLogger(LOGGER_NAME);
+
+        // when
+        simpleLogger.info(MESSAGE_FORMAT, PARAMETER);
+
+        // then
+        assertArrayListMatches(listener.capturedLogName, LOGGER_NAME);
+        assertArrayListMatches(listener.capturedLevel, LocationAwareLogger.INFO_INT);
+        assertArrayListMatches(listener.capturedMessageFormats, MESSAGE_FORMAT);
+        assertArrayListMatches(listener.capturedMessages, MESSAGE_FORMAT.replaceAll("\\{}", PARAMETER));
```
Comment (neeme-praks-sympower) [inline slf4j-simple/src/test/java/org/slf4j/impl/SimpleLoggerTest.java]: I personally do not consider such value manipulation to be a good style in unit tests. Better have a constant, a'la:
```java
private static final String EXPECTED_FORMATTED_MESSAGE = "This is a format: " + PARAMETER;
```
And then:
```suggestion
        assertArrayListMatches(listener.capturedMessages, EXPECTED_FORMATTED_MESSAGE);
```

**Review by neeme-praks-sympower — CHANGES_REQUESTED** [review]
Suggested some minor changes.

**Review by MikeNikolopoulos — CHANGES_REQUESTED** [review]
Neeme has covered me.

File: slf4j-simple/src/test/java/org/slf4j/impl/SimpleLoggerTest.java [hunk 3]
```diff
@@ -40,6 +40,7 @@
     private static final String MESSAGE_FORMAT = "This is a format: {}";
     private static final String PARAMETER = "This is an argument";
     private static final String LOGGER_NAME = "some.logger";
+    private static final String EXPECTED_FORMATTED_MESSAGE = "This is a format: " + PARAMETER;
```
Comment (neeme-praks-sympower) [inline slf4j-simple/src/test/java/org/slf4j/impl/SimpleLoggerTest.java:42]: I would suggest to move this right after `MESSAGE_FORMAT`, so it is more obvious that these two are related.

**Review by neeme-praks-sympower — APPROVED** [review]
One minor comment, can be approved already.

**Review by MikeNikolopoulos — APPROVED** [review]

**Pull request merged**